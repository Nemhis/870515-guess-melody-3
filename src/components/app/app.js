import React from 'react';
import WelcomeScreen from '../welcome-screen/welcome-screen';


const App = (props) => {
  // eslint-disable-next-line react/prop-types
  const {name} = props;

  return <React.Fragment>
    <WelcomeScreen name={name} />
  </React.Fragment>;
};

export default App;
