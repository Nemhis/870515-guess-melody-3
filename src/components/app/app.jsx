import React from 'react';
import PropTypes from 'prop-types';

import WelcomeScreen from '../welcome-screen/welcome-screen.jsx';

const App = (props) => {
  const {name} = props;

  return <React.Fragment>
    <WelcomeScreen name={name} onWelcomeButtonClick={() => {}}/>
  </React.Fragment>;
};

App.propTypes = {
  name: PropTypes.string,
};

export default App;
