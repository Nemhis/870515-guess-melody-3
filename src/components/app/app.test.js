import React from 'react';
import renderer from 'react-test-renderer';

import {App} from './app.jsx';

const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [{
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `blues`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `jazz`,
    }, {
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
      genre: `rock`,
    }],
  }, {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
    },
    answers: [{
      picture: `https://api.adorable.io/avatars/128/1`,
      artist: `John Snow`,
    }, {
      picture: `https://api.adorable.io/avatars/128/2`,
      artist: `Jack Daniels`,
    }, {
      picture: `https://api.adorable.io/avatars/128/3`,
      artist: `Jim Beam`,
    }],
  },
];

describe(`Render correctly <App>`, () => {
  it(`Render welcome screen`, () => {
    const tree = renderer
      .create(<App
        questions={questions}
        errorsCount={3}
        step={-1}
        onWelcomeButtonClick={() => {}}
        onUserAnswer={() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render genre screen`, () => {
    const tree = renderer
      .create(<App
        questions={questions}
        errorsCount={3}
        step={0}
        onWelcomeButtonClick={() => {}}
        onUserAnswer={() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Render artist screen`, () => {
    const tree = renderer
      .create(<App
        questions={questions}
        errorsCount={3}
        step={1}
        onWelcomeButtonClick={() => {}}
        onUserAnswer={() => {}}
      />, {
        createNodeMock: () => {
          return {};
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
