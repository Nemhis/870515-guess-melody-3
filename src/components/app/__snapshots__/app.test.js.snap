// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> Render <App> 1`] = `
<button
  className="welcome__button"
  onClick={[Function]}
>
  <span
    className="visually-hidden"
  >
    Начать игру
  </span>
</button>
`;
